File: src/list-sort.go
/*
#	What Is This: programming samples
#	Author: Makoto Takeshita <takeshita.sample@gmail.com>
#	URL: http://simplesandsamples.com
#	Version: UNBORN
#	
#	Usage:
#	 1. git clone https://github.com/takeshitamakoto/sss.git
#	 2. change the directory name to easy-to-use name. (e.g. sss -> sample)
#	 3. open sss/src/filename when you need any help.
#	
*/

package main
import "fmt"
import "sort"
type IntArray []int
func (p IntArray) Len() int           {return len(p);}
func (p IntArray) Less(i, j int) bool {return p[i] < p[j];}
func (p IntArray) Swap(i, j int)      {p[i], p[j] = p[j], p[i];}
func main() {
	data := []int{ 3,2,5,4,7,6,1 };
	a := IntArray(data);
	fmt.Println( a );
	sort.Sort(a);
	fmt.Println( a );
}

File: src/list-sort.erl
%
%	What Is This: programming samples
%	Author: Makoto Takeshita <takeshita.sample@gmail.com>
%	URL: http://simplesandsamples.com
%	Version: UNBORN
%	
%	Usage:
%	 1. git clone https://github.com/takeshitamakoto/sss.git
%	 2. change the directory name to easy-to-use name. (e.g. sss -> sample)
%	 3. open sss/src/filename when you need any help.
%	

main(_)->
           Lis = [3,2,5,4,7,6,1],
           io:format("~p~n",[Lis]),
           % ライブラリ使用の場合
           io:format("~p~n",[lists:sort(Lis)]),
           % 自作の場合
           io:format("~p~n",[mysort(Lis)]).
mysort([]) -> [];
mysort([H|T]) -> mysort([A || A<-T,  A < H])
                       ++ [H] ++
                 mysort([A || A<-T, A >= H]).

